src/httpServer.ts

```diff
@@ -170,26 +176,80 @@ function configureServer(httpServer: restify.Server) {
     })
 
-    httpServer.get('/api/v1.0/tasks/:id', async (req, res, next) => {
+    httpServer.get('/api/v1.0/me/tasks/:id', async (req, res, next) => {
         let id = req.params["id"];
         await graphGet(req, res, next, `https://graph.microsoft.com/beta/me/outlook/tasks/${id}?${app.graph.queryExpandNagExtensions}`);
     })
 
-    httpServer.patch('/api/v1.0/tasks/:id', async (req, res, next) => {
+    httpServer.patch('/api/v1.0/me/tasks/:id', async (req, res, next) => {
         let id = req.params["id"];
         let data = req.body;
         await graphPatch(req, res, next, `https://graph.microsoft.com/beta/me/outlook/tasks/${id}?${app.graph.queryExpandNagExtensions}`, data);
     })
 
-    httpServer.get('/api/v1.0/me', async (req, res, next) => {
-        await graphGet(req, res, next, "https://graph.microsoft.com/v1.0/me");
+    httpServer.get('/api/v1.0/me/connections', async (req, res, next) => {
+        let error: any;
+        try {
+            let accessToken = await app.authManager.getAccessTokenFromAuthKey(getCookie(req, 'userId'));
+            let conversations = await app.graph.getConversations(accessToken);
+            res.json(200, conversations);
+            res.end();
+            return next();
+        }
+        catch (err) {
+            error = err;
+        }
+        res.status(400);
+        res.json({ error });
+        res.end();
+        return next();
     })
 
+    httpServer.patch('/api/v1.0/me/connections/:id', async (req, res, next) => {
+        let id = req.params["id"];  // this is ignored for now
+        let data = req.body;
+        let error: any;
+        try {            
+            let authContext = await app.authManager.getAuthContextFromAuthKey(getCookie(req, 'userId'));
+            if (!authContext || !authContext.oid) throw new Error('/me/connections-PATCH: Could not identify user');
+            app.conversationManager.upsert(authContext.oid, data);
+            res.status(200);
+            res.end();
+            return next();
+        } catch (err) { error = error }
+        res.status(400);
+        res.json({ error });
+        res.end();
+        return next();
+    })
+
+    httpServer.del('/api/v1.0/me/connections/:id', async (req, res, next) => {
+        let id = req.params["id"];  // this is ignored for now
+        let data = req.body;
+        let error: any;
+        try {            
+            let authContext = await app.authManager.getAuthContextFromAuthKey(getCookie(req, 'userId'));
+            if (!authContext || !authContext.oid) throw new Error('/me/connections-PATCH: Could not identify user');
+            app.conversationManager.delete(authContext.oid, data);
+            res.status(200);
+            res.end();
+            return next();
+        } catch (err) { error = error }
+        res.status(400);
+        res.json({ error });
+        res.end();
+        return next();
+    })
 
     /// Interactive Tests
```

Comment:
what are interactive tests? is their test specific code in the httpserver?

Reply:
To be removed